src/app-layer-htp-range.c

```diff
@@ -545,14 +545,21 @@ File *HttpRangeClose(const StreamingBufferConfig *sbcfg, HttpRangeContainerBlock
             // a new range just begins where we ended, append it
             if (range->gap > 0) {
                 // if the range had a gap, begin by it
-                if (FileAppendData(c->container->files, sbcfg, NULL, range->gap) != 0) {
+                uint32_t gap = range->gap <= UINT32_MAX ? (uint32_t)range->gap : UINT32_MAX;
+                if (FileAppendData(c->container->files, sbcfg, NULL, gap) != 0) {
                     c->container->lastsize = f->size;
                     HttpRangeFileClose(sbcfg, c->container, flags | FILE_TRUNCATED);
                     c->container->error = true;
                     return f;
                 }
             }
-            if (FileAppendData(c->container->files, sbcfg, range->buffer, range->offset) != 0) {
+            if (range->offset > UINT32_MAX) {
```

Comment:
does this mean we can't support ranges for files > 4GiB?

Reply:
No, that means we do not support ranges >4GiB for files

For example, We support a 15 GiB file if it is split into 15 1GiB ranges (but util-file.c has the `DEBUG_VALIDATE_BUG_ON(data_len > BIT_U32(26)); // 64MiB as a limit per chunk seems already excessive` in this case)
We do not support this 15 GiB file if it is split into 3 5GiB files, because `FileAppendData` only accepts a u32